scripts/linters/codeowner_linter.py

```diff
@@ -147,27 +153,31 @@ def check_for_important_patterns_at_bottom_of_codeowners(important_patterns):
     list_minus_critical_rule_section_set = (
         codeowner_important_paths_set.difference(important_patterns_set))
     for rule in critical_rule_section_minus_list_set:
-        python_utils.PRINT(
+        summary_message = (
             '%s --> Rule %s is not present in the '
             'CODEOWNER_IMPORTANT_PATHS list in '
             'scripts/linters/pre_commit_linter.py. Please add this rule in the '
             'mentioned list or remove this rule from the \'Critical files'
             '\' section.' % (CODEOWNER_FILEPATH, rule))
+        python_utils.PRINT(summary_message)
+        summary_messages.append(summary_message)
         failed = True
     for rule in list_minus_critical_rule_section_set:
-        python_utils.PRINT(
+        summary_message = (
             '%s --> Rule \'%s\' is not present in the \'Critical files\' '
             'section. Please place it under the \'Critical files\' '
             'section since it is an important rule. Alternatively please '
             'remove it from the \'CODEOWNER_IMPORTANT_PATHS\' list in '
             'scripts/linters/pre_commit_linter.py if it is no longer an '
             'important rule.' % (CODEOWNER_FILEPATH, rule))
+        python_utils.PRINT(summary_message)
+        summary_messages.append(summary_message)
         failed = True
 
-    return failed
+    return failed, summary_messages
 
 
-def check_codeowner_file(verbose_mode_enabled):
+def check_codeowner_file(file_cache, verbose_mode_enabled):
```

Comment:
Any specific reason for this change?

Reply:
we are using a global variable across modules, I think that may not be the best practice to use a variable like that. Tests are also not running with that configuration.

Reply:
There must be some intent behind that configuration can you please git blame that line and talk to the author about this changes?

Reply:
I am the one who added that line, when I splitted the linter in different modules.